src/mlpack/methods/ann/layer/layer_types.hpp

```diff
@@ -92,6 +109,7 @@ using LayerTypes = boost::variant<
     DropConnect<arma::mat, arma::mat>*,
     Dropout<arma::mat, arma::mat>*,
     ELU<arma::mat, arma::mat>*,
+    FFN<NegativeLogLikelihood<>, RandomInitialization>*,
```

Comment:
It seems like this has to be added here so that you can pass a single layer or an entire `FFN` to the constructor of `NeuralTuringMachine`.  But unfortunately this means there is a restriction, that `NegativeLogLikelihood<>` and `RandomInitialization` must be used.  I have a different idea which maybe could be more helpful and robust; let me know what you think.

Instead of adding `FFN<>` to `LayerTypes`, add an additional constructor to `NeuralTuringMachine` that takes some `FFN` type (I guess a template type) and extracts the layers from it and adds them to `NeuralTuringMachine::network`.  I think we might have to add some functions to `FFN` so that individual layers could be accessed (not sure about that).